hw/drivers/sensors/bma253/src/bma253.c

```diff
@@ -4593,8 +4710,25 @@ bma253_init(struct os_dev * dev, void * arg)
         return rc;
     }
 
+#if MYNEWT_VAL(BMA253_INT_ENABLE)
     init_interrupt(&bma253->intr, bma253->sensor.s_itf.si_ints);
 
+    priv_data = os_malloc(sizeof(*priv_data));
```

Comment:
I would have made this a structure(pdd : per driver data) itself in the driver instead of a pointer. That's how I have done some of the other drivers. but that's just me. Malloc can fail at times.